js/app.js

```diff
@@ -56,6 +51,69 @@ handleMobileNav = function() {
   });
 };
 
+// Handle JSON
+Project.loadAll = function(dataWePassIn) {
+  JSON.parse(dataWePassIn).forEach(function(ele) {
+    projects.push(new Project(ele));
+  });
+};
+
+Project.appendAll = function(dataToAppend) {
+  dataToAppend.forEach(function(p){
+    $('#projects').append(p.toHtml());
+  });
+};
+
+Project.fetchAll = function(){
+  if (localStorage.data) {
+    console.log('Local Storage DOES Exist');
+
+    $.ajax({
+      url: 'data/projects-data.json',
+      success: function (data, message, xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        console.log('Current eTag = ' + eTag);
+        var compareETag = JSON.parse(localStorage.getItem('eTag1'));
+        console.log('Compare eTag = ' + compareETag);
+        // console.log(message, eTag);
+        if (eTag !== compareETag) {
+          console.log('Changes to JSON file detected');
+          console.log('Retrieved data from JSON file...');
+          Project.loadAll(JSON.stringify(data));
+          Project.appendAll(projects);
+          //Set Local Storage
+          localStorage.setItem('data', JSON.stringify(projects));
+          console.log('Setting Local Data...');
+          Project.appendAll(projects);
+        }
+        else {
+          console.log('NO change to JSON Data');
+          var localData = localStorage.getItem('data');
+          console.log('Retriving Local Data');
+          Project.loadAll(localData);
+          //Append to HTML
+        }
+        Project.appendAll(projects);
+      }
+    });
+  }
+  else {
+    $.getJSON('data/projects-data.json', function(data){
```

Comment:
You can get the etag here by just passing in data msg, xhr to the callback function. Once you have the xhr you have the eTag.